pkg/shared/clients/redis/redis_reader.go

```diff
@@ -94,11 +95,14 @@ func (br *RedisStreamsRead) Read(_ context.Context, count int64) ([]*isb.ReadMes
 		}
 	}
 
-	// for each XMessage in []XStream
-	msgs, err := br.XStreamToMessages(xstreams, messages, labels)
-	if br.Metrics.ReadsAdd != nil {
-		br.Metrics.ReadsAdd(len(messages))
+	// Update metric for number of messages read in
+	for _, xstream := range xstreams {
+		if br.Metrics.ReadsAdd != nil {
+			br.Metrics.ReadsAdd(len(xstream.Messages))
+		}
```

Comment:
this is technically the actual number of messages read in. The old code represents the number of messages published which theoretically could be different.